UI/src/view/point_asset/massTransitMarker.js

```diff
@@ -146,23 +154,82 @@
       return cacheImage(cachedImageKey, image, canvas.width, canvas.height );
     };
 
+    function extractServicePointsAuxiliarsValues() {
+      var resultValues = {palvelu: undefined, tarkenne: undefined};
+      var auxiliarData = data.payload ? data.payload.properties : data.propertyData;
+
+      if (auxiliarData !== undefined) {
+        resultValues.palvelu = _.find(auxiliarData, function(prop) {return prop.publicId == "palvelu";});
+        resultValues.tarkenne = _.find(auxiliarData, function(prop) {return prop.publicId == "tarkenne";});
+      }
+
+      if (resultValues.palvelu !== undefined) {
+        if (resultValues.palvelu.values.length > 0) {
+          resultValues.palvelu = resultValues.palvelu.values[0].propertyValue;
+        } else {
+          resultValues.palvelu = undefined;
+        }
+      }
+
+      if (resultValues.tarkenne !== undefined ) {
+        if (resultValues.tarkenne.values.length > 0) {
+          resultValues.tarkenne = resultValues.tarkenne.values[0].propertyValue;
+        } else {
+          resultValues.tarkenne = undefined;
+        }
+      }
+
+      return resultValues;
+    }
+
     var createStopTypeStyles = function(stopTypes, margin){
       var groupOffset = groupOffsetForAsset();
       var imgMargin = margin ? margin : 0;
       stopTypes.sort();
       var i = 0;
-      return _.map(_.isEmpty(stopTypes) ? [EMPTY_IMAGE_TYPE] : stopTypes, function(stopType) {
-        i++;
-        return new ol.style.Style({
-          image: new ol.style.Icon(({
-            anchor: [-(IMAGE_PADDING+1+imgMargin), (i * IMAGE_HEIGHT)+ IMAGE_PADDING + STICK_HEIGHT + imgMargin + groupOffset],
-            anchorXUnits: 'pixels',
-            anchorYUnits: 'pixels',
-            src: 'images/mass-transit-stops/' + stopType + '.png',
-            scale: styleScale
-          }))
+
+      if (stopTypes.length > 0 && stopTypes[0] == 7) {
+
+        var auxServicePointInfo = extractServicePointsAuxiliarsValues();
+        var srcImg;
+
+        if (auxServicePointInfo.palvelu !== "11") {
+          srcImg = SERVICE_POINT_IMAGES.find( function(spi) { if (spi.value == auxServicePointInfo.palvelu) return spi;});
+          srcImg = srcImg === undefined ? [EMPTY_IMAGE_TYPE]: srcImg.imgUrl;
+        }
+        else if (auxServicePointInfo.tarkenne !== undefined ) {
+            srcImg = SERVICE_POINT_IMAGES.find( function(spi) { if (spi.value == auxServicePointInfo.tarkenne ) return spi;});
+            srcImg = srcImg === undefined ? [EMPTY_IMAGE_TYPE]: srcImg.imgUrl;
+        }
+
+        return _.map(_.isEmpty(stopTypes) ? [EMPTY_IMAGE_TYPE] : stopTypes, function (stopType) {
```

Comment:
This _.isEmpty(stopTypes) in this line does not make sense since the if condition already checks this

Reply:
Already solved